core/src/main/java/org/elasticsearch/action/fieldstats/TransportFieldStatsTransportAction.java

```diff
@@ -119,18 +119,14 @@ protected FieldStatsResponse newResponse(FieldStatsRequest request, AtomicRefere
                 while (iterator.hasNext()) {
                     Map.Entry<String, Map<String, FieldStats>> entry = iterator.next();
                     FieldStats indexConstraintFieldStats = entry.getValue().get(indexConstraint.getField());
-                    if (indexConstraintFieldStats == null) {
```

Comment:
We filter it if the index constraint doesn't match. If it is null it means that no document has a value for the index where the index constraint is defined for, so we filter it out too (it will never match with the index constraint)